### PR TITLE
fix: upgrade `fuels-code-gen` to 0.49

### LINKED ISSUE #1484: Errors "expected `AssetId`, found `[u8; 32]`  arguments to this function are incorrect" and "Unrecognized primitive type: `str`." during `forc index build`
Hi guys 

I tried to build an indexer with abi of one of our spark perp contracts and got this error 
image

after I removed str from abi, and after that I got that error (2nd screenshot)
image

## my toolchain
```
Default host: aarch64-apple-darwin
fuelup home: /Users/alexey/.fuelup

installed toolchains
--------------------
beta-3-aarch64-apple-darwin
beta-4-rc.2-aarch64-apple-darwin
latest-aarch64-apple-darwin
beta-4-aarch64-apple-darwin (default)
hotfix
latest-2023-09-25-aarch64-apple-darwin
latest-2023-11-15-aarch64-apple-darwin
my-custom-toolchain

active toolchain
-----------------
beta-4-aarch64-apple-darwin (default)
  forc : 0.46.1
    - forc-client
      - forc-deploy : 0.46.1
      - forc-run : 0.46.1
    - forc-doc : 0.46.1
    - forc-explore : 0.28.1
    - forc-fmt : 0.46.1
    - forc-index : 0.23.0
    - forc-lsp : 0.46.1
    - forc-tx : 0.46.1
    - forc-wallet : 0.3.0
  fuel-core : 0.20.5
  fuel-core-keygen : Error getting version string
  fuel-indexer : 0.23.0

fuels versions
---------------
forc : 0.45
forc-wallet : 0.45
```

here is my abi
[account-balance-abi.json](https://github.com/FuelLabs/fuel-indexer/files/13506648/account-balance-abi.json)

### PR DESCRIPTION
**Pull request by ra0x3**
Thanks for opening a PR with the Fuel Indexer project. Please review the **Checklist** below and ensure you've completed all of the necessary steps to make this PR review as painless as possible.

### Checklist
- [ ] Ensure your top-level commit message is in line with our [contributor guidelines](./CONTRIBUTING.md).
- [ ] Please add proper labels.
- [ ] If there is an issue associated with this PR, please link the issue (right-hand sidebar)
- [ ] If there is not an issue associated with this PR, add this PR to the "Fuel Indexer" project (right-hand sidebar)
- [ ] Please allow Codeowners at least 24 hours to do a first-pass review.
- [ ] Please add thoroughly detailed testing steps below.
- [ ] Please keep your Changelog message short and sweet.

### Description

- Fixes bug reported by SwayGang related to type name collisions
- Resolution was discussed in https://github.com/FuelLabs/fuel-indexer/issues/1487
- TLDR: Had to upgrade `fuels-code-gen` to `>= 0.49`

### Testing steps

- [ ] Since `configurables` were added to test indexer, it should compile with no errors

### Changelog

- fix: upgrade fuels-code-gen to 0.49

### CODE REVIEW

File: packages/fuel-indexer-tests/contracts/fuel-indexer-test/out/debug/fuel-indexer-test-abi.json [hunk 1]
```diff
@@ -262,6 +262,18 @@
     },
     {
       "typeId": 21,
+      "type": "struct AssetId",
```
Comment (ra0x3) [inline packages/fuel-indexer-tests/contracts/fuel-indexer-test/out/debug/fuel-indexer-test-abi.json:265]: @segfault-magnet Here's the `AssetId` added to the ABI JSON. I planned to just skip these when pulling these types into the indexer (to avoid the name collision) but that doesn't work (due to my comment below)

File: packages/fuel-indexer-tests/contracts/fuel-indexer-test/src/main.sw [hunk 2]
```diff
@@ -15,6 +15,12 @@ use std::{
     token::*,
 };
 
+configurable {
```
Comment (ra0x3) [inline packages/fuel-indexer-tests/contracts/fuel-indexer-test/src/main.sw:18]: @segfault-magnet Here's the usage of `configurables` that SwayGang had (the link that was giving you a 404)

**Review by lostman — APPROVED** [review]

**Review by deekerno — APPROVED** [review]
